ngrep.c

```diff
@@ -105,9 +106,10 @@
  */
 
 uint16_t snaplen = 65535, limitlen = 65535, promisc = 1, to = 100;
-uint16_t match_after = 0, keep_matching = 0, matches = 0, max_matches = 0;
+unsigned long match_after = 0, keep_matching = 0, matches = 0, max_matches = 0;
+
 #if USE_TCPKILL
-uint16_t tcpkill_active = 0;
+unsigned long tcpkill_active = 0;
```

Comment:
This is a bad idea. `tcpkill_active` is used as the third parameter to a routine from ported from dSniff called [`tcpkill_kill`](https://github.com/haguenau/ngrep/commit/f2a1f8b09194a8ab87787c1fdb60b968f5957c05#diff-a3cce3c8aad81bc94746a67bc7a0b5e2R54). It expects the third parameter to be `unsigned`, but ultimately gets compared against an `int`. So the real max value that can be passed with -K is `sizeof(int)` (aka `uint16_t`).

Probably best to leave this one as is.